### PR TITLE
Fix test_show_collection_number [#173205465]

### PR DESCRIPTION
**Pull request by JoeCohen**
This is the 1st commit in a branch intended to get CollectionNumbersControllerTest to pass.

### CODE REVIEW

File: app/helpers/object_link_helper.rb [hunk 1]
```diff
@@ -7,7 +7,7 @@ module ObjectLinkHelper
     Google_Maps: "https://maps.google.com/maps?q=",
     Google_Search: "https://www.google.com/search?q=",
     Wikipedia: "https://en.wikipedia.org/w/index.php?search="
-  }.freeze
+  }
```
Comment (JoeCohen) [inline app/helpers/object_link_helper.rb:10]: Redundant (and offend RuboCop) in light of magic comment at top of file: `# frozen_string_literal: true`

File: app/controllers/collection_numbers_controller.rb [hunk 2]
```diff
@@ -17,23 +17,23 @@ class CollectionNumbersController < ApplicationController
...
   # Show list of collection_numbers.
-  def index # :norobots:
+  def index
     store_location
     query = create_query(:CollectionNumber, :all)
     show_selected_collection_numbers(query)
   end
 
-  alias_method :list_collection_numbers, :index
+  alias list_collection_numbers index
```
Comment (JoeCohen) [inline app/controllers/collection_numbers_controller.rb:33]: RuboCop offense. See https://docs.rubocop.org/rubocop/cops_style.html#stylealias
and
https://rubystyle.guide/#alias-method-lexically

File: app/controllers/collection_numbers_controller.rb [hunk 3]
```diff
@@ -97,19 +97,28 @@ def new # :norobots:
     redirect_back_or_default("/") unless @collection_number
   end
 
-  alias_method :create_collection_number, :new
+  alias create_collection_number new
 
   def create
+    store_location
```
Comment (JoeCohen) [inline app/controllers/collection_numbers_controller.rb:103]: Adds lines that appeared in master.

File: app/views/collection_numbers/_form.html.erb [hunk 4]
```diff
@@ -1,7 +1,6 @@
-<%= # form_tag(add_query_param(action: :new, id: @observation.id)) do %>
-<%= form_with model: :collection_number do |f| %>
+<%= form_with model: @collection_number do |f| %>
 
-  <%= submit_tag(button.t, class: "btn mx-auto mt-3") %>
+  <%= submit_tag(local_assigns[:button].t, class: "btn mx-auto mt-3") %>
```
Comment (JoeCohen) [inline app/views/collection_numbers/_form.html.erb:3]: The combination of `locals: { button: :SAVE }` in the view and `button` in the form was not working (for reasons I don't understand.

File: test/controllers/collection_numbers_controller_test.rb [hunk 5]
```diff
@@ -3,7 +3,7 @@
 require "test_helper"
 
 class CollectionNumbersControllerTest < FunctionalTestCase
-  def test_collection_index
+  def test_index
```
Comment (JoeCohen) [inline test/controllers/collection_numbers_controller_test.rb:6]: This shortens names and uses the 7 CRUD action names.
See NIMMO TODO at former line 65.

**Comment by JoeCohen** [comment]
I'd like to get this merged although there are still 2 failures; I may not have serious time to code until ~Jun 16 (and I will be off the grid during much of that time).
It's an improvement, and merging it sooner will reduce the odds of conflicts.

**Comment by JoeCohen** [comment]
The 2 failures are quoted below. I'm stumped by both of them.
```
vagrant@vagrant /vagrant/mushroom-observer (collection_nos_ctlr_test-#173205465)
$ rails test test/controllers/collection_numbers_controller_test.rb
Started with run options --seed 59822

 FAIL["test_create_redirect", #<Minitest::Reporters::Suite:0x0000562e2ca8ad08 @name="CollectionNumbersControllerTest">, 25.48611131600046]
 test_create_redirect#CollectionNumbersControllerTest (25.49s)
        Expected at least 1 element matching "form[action*='observations/432936842?q=1V']", found 0..
        Expected 0 to be >= 1.
        test/controllers/collection_numbers_controller_test.rb:213:in `test_create_redirect'

 FAIL["test_edit_redirect", #<Minitest::Reporters::Suite:0x0000562e2debeec8 @name="CollectionNumbersControllerTest">, 25.605426284000714]
 test_edit_redirect#CollectionNumbersControllerTest (25.61s)
        Expected at least 1 element matching "form[action *= 'collection_number/461709293?back=foo&q=1W']", found 0..
        Expected 0 to be >= 1.
        test/controllers/collection_numbers_controller_test.rb:342:in `test_edit_redirect'

  22/22: [====================================================================================================] 100% Time: 00:00:28, Time: 00:00:28

Finished in 28.32966s
22 tests, 217 assertions, 2 failures, 0 errors, 0 skips
Coverage report generated for Unit Tests to /vagrant/mushroom-observer/coverage. 4787 / 11317 LOC (42.3%) covered.
```

**Comment by pellaea** [comment]
I'd recommend dumping the @response.body right before the assertion to see
what Rails is actually outputting.  For example, the new forms may be
putting the q=IV (query_params) into form data instead of the action URL.
Who knows?  So much has changed so subtly!

**Comment by JoeCohen** [comment]
1. In `test_edit_redirect`, the query params are actually getting to the right place. The problem is `@back_object`, whichs point to an observation, instead of a collection number. I'm actually befuddled how it worked in master.
2. In `test_create_redirect`, the test assertion is not finding a form.
3. I'll try your suggestion in about a week -- no time until then. Right now I have to see if I can quickly fix all the conflcts.

**Comment by JoeCohen** [comment]
Please ignore this PR for now; I'm converting it back to a draft. 
There are too many conflicts for me to deal with tonite. (And maybe resolving them will help with the tests.)

**Comment by JoeCohen** [comment]
I am closing this PR. It is replaced by Draft PR #571

**Comment by JoeCohen** [comment]
I finally figured out that the thing that was hangin me up was:
`<% render "form" ...` which should have been
`<%= render "form" ...`
Fixing that (in #571) revealed the other problems (also now fixed in #571).

**Pull request closed without merging**